### PR TITLE
Parse PolyStyle.

### PR DESCRIPTION
**Pull request by EBartusiak**
- Parse PolyStyle.
- If boolean values for <fill> or <outline> are false, then the corresponding color element is set to be transparent (to make them invisible).

### CODE REVIEW

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/styles/PolyStyle.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    mFillColor = fillColor;
+  }
+
+  public int getFillColor() {
+    return mFillColor;
+  }
+
+  public int getTransparent() {
+    return 0x00ffffff;
+  }
```
Comment (adayli) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/styles/PolyStyle.java:40]: Can this be a static constant instead?

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 2]
```diff
@@ -133,48 +136,53 @@ private void parseStyleAddToMapStylesHolders()
...
       if (mParser.getEventType() != XmlPullParser.START_TAG) {
         continue;
       }
-      String type = mParser.getName();
-      if (type.equals("href")) {
+      if (mParser.getName().equals("href")) {
         String url = parseText();
         InputStream inputStream = new URL(url).openConnection().getInputStream();
-        stylesHolder.getIconStyle().setImage(mFactory.createMapImage(inputStream));
+        iconStyle.setImage(mFactory.createMapImage(inputStream));
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:180]: The icon style mutating when parsing the icon seems wonky. Can you have two icons with the same style but with different hrefs on each icon?

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 3]
```diff
@@ -196,15 +204,62 @@ private void parseLineStyle(@NonNull StylesHolder stylesHolder)
...
+  private void parsePolyStyle(@NonNull PolyStyle polyStyle)
+      throws XmlPullParserException, IOException, KMLParseException {
+    while (moveToNext() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      switch (mParser.getName()) {
+        case "fill":
+          String shouldFill = parseText();
+          if (shouldFill.equals("1") || shouldFill.equalsIgnoreCase("true")) {
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:225]: Don't copy/paste this; create a common function to parse a boolean.

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 4]
```diff
@@ -196,15 +204,62 @@ private void parseLineStyle(@NonNull StylesHolder stylesHolder)
...
+            throw new KMLParseException(
+                "Invalid value ("
+                    + shouldOutline
+                    + ") for <outline> at position "
+                    + mParser.getPositionDescription());
+          }
+          break;
+        case "color":
+          if (polyStyle.shouldFill()) {
+            polyStyle.setFillColor(parseColor());
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:253]: This seems wrong. What if fill appears after the color tag?

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 5]
```diff
@@ -558,6 +613,21 @@ private void applyStyles() throws KMLParseException {
...
           line.setStrokeWidth(stylesHolder.getLineStyle().getWidth());
           line.setStrokeColor(stylesHolder.getLineStyle().getStrokeColor());
+        } else if (element instanceof MapPolygon) {
+          MapPolygon polygon = (MapPolygon) element;
+          LineStyle lineStyle = stylesHolder.getLineStyle();
+          PolyStyle polyStyle = stylesHolder.getPolyStyle();
+          if (polyStyle.shouldOutline()) {
+            polygon.setStrokeColor(lineStyle.getStrokeColor());
+          } else {
+            polygon.setStrokeColor(polyStyle.getTransparent());
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:623]: Can these two lines be replaced with setting stroke width to 0?

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java (hunk 4 above)
Comment (EBartusiak) [reply Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:253]: Changed this to always set fillColor to what is specified while parsing. When applying the styles, the fillColor of the polygon is only set if polyStyle.shouldFill() is true (else fillColor is set to transparent).

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java (hunk 2 above)
Comment (EBartusiak) [reply Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:180]: Resolved in Teams: Icon is an element under IconStyle. Icon may have an href element that describes the image to be used for the MapIcon. Point is an element under Placemark that describes the MapIcon for the map.

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 6]
```diff
@@ -489,6 +520,19 @@ private String parseText() throws IOException, XmlPullParserException, KMLParseE
     }
   }
 
+  private boolean parseBooleanIsTrue()
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:523]: parseBoolean is plenty descriptive.

**Pull request merged**